iron-scroll-target-behavior.html

```diff
@@ -0,0 +1,216 @@
+
+          this._boundScrollHandler = this._boundScrollHandler || this._scrollHandler.bind(this);
+          // Add a new listener
+          if (scrollTarget === this._doc) {
+            window.addEventListener('scroll', this._boundScrollHandler);
+            if (this._scrollTop !== 0 || this._scrollLeft !== 0) {
+              this._scrollHandler();
+            }
+          } else if (scrollTarget && scrollTarget.addEventListener) {
+            scrollTarget.addEventListener('scroll', this._boundScrollHandler);
+          }
+          this._oldScrollTarget = scrollTarget;
+        }
+      }
+    },
+
+   /**
+    * Runs on every scroll event. Consumer of this behavior may want to override this method.
+    *
+    * @protected
+    * @abstract
+    */
+    _scrollHandler: function scrollHandler() {},
+
+   /**
+    * The default scroll target. Consumers of this behavior may want to customize
+    * the default scroll target.
+    *
+    * @type {Element}
+    */
+    get _defaultScrollTarget() {
+      return this._doc;
+    },
+
+   /**
+    * Shortcut for the document element
+    *
+    * @type {Element}
+    */
+    get _doc() {
+      return this.ownerDocument.documentElement;
+    },
+
+   /**
+    * Gets the number of pixels that the content of an element is scrolled upward.
+    *
+    * @type {number}
+    */
+    get _scrollTop() {
+      if (this.scrollTarget) {
+        return this.scrollTarget === this._doc ? window.pageYOffset : this.scrollTarget.scrollTop;
+      }
+      return 0;
+    },
+
+   /**
+    * Gets the number of pixels that the content of an element is scrolled to the left.
+    *
+    * @type {number}
+    */
+    get _scrollLeft() {
+      if (this.scrollTarget) {
+        return this.scrollTarget === this._doc ? window.pageXOffset : this.scrollTarget.scrollLeft;
+      }
+      return 0;
+    },
+
+    /**
+     * Sets the number of pixels that the content of an element is scrolled upward.
+     *
+     * @type {number}
+     */
+    set _scrollTop(top) {
+      if (this.scrollTarget === this._doc) {
+        window.scrollTo(window.pageXOffset, top);
+      } else if (this.scrollTarget) {
+        this.scrollTarget.scrollTop = top;
+      }
+    },
+
+    /**
+     * Sets the number of pixels that the content of an element is scrolled to the left.
+     *
+     * @type {number}
+     */
+    set _scrollLeft(left) {
+      if (this.scrollTarget === this._doc) {
+        window.scrollTo(left, window.pageYOffset);
+      } else if (this.scrollTarget) {
+        this.scrollTarget.scrollLeft = left;
+      }
+    },
+
+    /**
+     * Scrolls the content to a particular place.
+     *
+     * @method scroll
+     * @param {number} top The top position
+     * @param {number} left The left position
+     */
+    scroll: function(top, left) {
+       if (this.scrollTarget === this._doc) {
+        window.scrollTo(top, left);
+      } else if (this.scrollTarget) {
+        this.scrollTarget.scrollTop = top;
```

Comment:
Since you accept `String` types for `scrollTarget`, and since the observer that changes it to an `HTMLElement` isn't fired unless you are also attached, isn't there a danger here that `scroll` might be called when `scrollTarget` is still a string?